crates/core/src/client/mod.rs

```diff
@@ -342,7 +355,7 @@ impl<P: Provider + Send + Sync> KakarotEthApi<P> for KakarotClient<P> {
 
                     TransactionReceipt {
                         transaction_hash,
-                        transaction_index: None,
+                        transaction_index: Some(U256::from(0)),
```

Comment:
I double checked and this is the only change needed to get a deploy working initially. 

High level, the `forge` script is backed by an ethers-rs client. The main category of bug we had here was the ethers client was trying to deserialize our responses and serde was throwing an error.